pages/index.tsx

```diff
@@ -1,51 +1,21 @@
 import React from 'react'
+import camelCase from 'lodash/camelCase'
+import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
 
 import { fetchContent } from '../services/contentful'
 import { LayoutFramework } from '../components/layouts/Framework'
 import { Hero } from '../components/presenters/Hero'
-import {
-  ContentPanel,
-  LeftCol,
-  RightCol,
-} from '../components/presenters/ContentPanel'
+import { ContentPanel, LeftCol } from '../components/presenters/ContentPanel'
 
-/**
- * Map content types to components and fields to props
- *
- */
-function getPresenter(type, fields) {
-  const componentMap = {
-    Hero: (
-      <Hero id="" title={fields.title} description={fields.heroDescription} />
-    ),
-    ContentBlock: (
-      <ContentPanel id="">
-        <LeftCol>LeftCol</LeftCol>
-        <RightCol>RightCol</RightCol>
-      </ContentPanel>
-    ),
-    // ...
-  }
-
-  return componentMap[type]
-}
-
-export default function Home({ sections }) {
-  const articleCollection = sections.flatMap(
-    (item) => item.articleCollection.items
-  )
-
-  const contentCollection = articleCollection.flatMap(
-    (item) => item.contentCollection.items
-  )
+export type ContentType =
+  | 'Hero'
```

Comment:
I wonder if rather than using these strings we could do `Component.name` e.g. `Hero.name`

Reply:
These are Contentful content types - they will be auto generated in upcoming PR by graphql-codegen.